# Planning Report

## Part 1: Data Description

In [35]:
library(tidyverse)
library(tidymodels)

In [36]:
players <- read_csv("https://raw.githubusercontent.com/arian-d1/dsci100_planning_report/refs/heads/main/data/players.csv")
sessions <- read_csv("https://raw.githubusercontent.com/arian-d1/dsci100_planning_report/refs/heads/main/data/sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [3]:
dim(players)

[1] 196   7

In [4]:
dim(sessions)

[1] 1535    5

In [5]:
head(sessions)

hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

In [6]:
head(players)

experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

In [7]:
summary(players)

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

In [8]:
players |>
    group_by(experience) |>
    summarize(count = n())

experience count
1 Amateur    63   
2 Beginner   35   
3 Pro        14   
4 Regular    36   
5 Veteran    48

In [9]:
players |>
    group_by(gender) |>
    summarize(count = n())

gender            count
1 Agender             2  
2 Female             37  
3 Male              124  
4 Non-binary         15  
5 Other               1  
6 Prefer not to say  11  
7 Two-Spirited        6

In [33]:
sessions_mutated <- sessions |>
    mutate(across(start_time:end_time, as_datetime)) |>
    mutate(time_spent_secs = round(as.numeric(end_time - start_time)), 2) |>
    arrange(desc(time_spent_secs))
    
sessions_mutated <- sessions_mutated |>
    mutate(time_spent_hours = round(time_spent_secs / 3600), 2)

head(sessions_mutated)

hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
3 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
4 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
  start_time          end_time            original_start_time original_end_time
1 2027-06-21 00:22:21 2028-06-21 00:02:40 1.71953e+12         1.71954e+12      
2 2003-08-21 00:21:36 2004-08-21 00:01:51 1.72272e+12         1.72274e+12      
3 2003-08-21 00:22:33 2004-08-21 00:00:51 1.72272e+12         1.72273e+12      
4 2019-06-21 00:22:44 2020-06-21 00:00:19 1.71884e+12         1.71884e+12      
5 2015-06-21 00:23:25 2016-06-21 00:00:52 1.71849e+12         1.71850e+12      
6 2011-08-21 00:23:39 2012-08-21 00:01:05 1.72342e+12         1.72342e+12      
  time_spent_secs 2 time_spent_hours
1 31621219        2 8784            
2 31621215        2 8784            
3 31621098        2 8784            
4 31621055        2 8784            
5 31621047        2 8784            
6 31621046        2 8784

In [34]:
sessions_mutated |>
    select(time_spent_hours) |>
    summarize(mean_hours = round(mean(time_spent_hours, na.rm = TRUE), 2))

mean_hours
1 7.01

Provide a full descriptive summary of the dataset, including information such as the number of observations, summary statistics (report values to 2 decimal places), number of variables, name and type of variables, what the variables mean, any issues you see in the data, any other potential issues related to things you cannot directly see, how the data were collected, etc. Make sure to use bullet point lists or tables to summarize the variables in an easy-to-understand format.

| Variable | Type | Meaning | Issues in Dataset | Non-visible Issues |
|----------|------|---------|-------------------|--------------------|
| experience | Character | Player’s experience level (Beginner, Amateur, Regular, Veteran, Pro) | N/A | May be self-reported, not objectively measured |
| subscribe | Logical | Indicates if the player has an active subscription | Major imbalance (majority are TRUE) | Subscription status may not directly reflect engagement |
| hashedEmail | Character | Anonymized unique player identifier | N/A | N/A |
| played_hours | Numeric | Total hours a player has played | The majority of players have very few, if none, hours played | A few players dominate the averages |
| name | Character | Player’s first name | N/A | N/A |
| gender | Character | Player’s gender (e.g., Male, Female) | N/A | May introduce bias for those who specified vs those who did not|
| Age | Numeric | Player’s age in years | Some values are na | If it's self-reported there may be inconsistencies |


There are 196 unique players in the players dataset in total. The players dataset has 7 variables: experience, subscribe, hashedEmail, played_hours, name, gender, and Age.
  
The mean played_hours is 5.85 hours. The max and min are 223.10 and 0 hours, respectively.  
The mean age is 21.14 hours. The max and min are 58 and 9 years old, respectively.  
The five specified experience classes are Beginner, Amateur, Regular, Veteran, and Pro.  
The 7 specified genders options are Agender, Female, Male, Non-binary, Other, Prefer not to say, Two-Spirited.

| Variable | Type | Meaning | Issues in Dataset | Non-visible Issues |
|----------|------|---------|-------------------|--------------------|
| hashedEmail | Character | Identifier linking session to player | None visible | N/A |
| start_time | Character | Timestamp when session began | N/A | N/A |
| end_time | Character | Timestamp when session ended | N/A | Some sessions may be logged incorrectly if app crashed or disconnected |
| original_start_time | Double | Original unadjusted session start time | N/A | N/A |
| original_end_time | Double | Original unadjusted session end time | N/A | N/A |

There are 1585 unique player sessions in the sessions dataset. The sessions datafame has 5 variables: hashedEmail, start_time, end_time, original_start_time, and original_end_time.  

The mean session length is 7.01 hours.

## Part 2: Questions

## Part 3: Exploratory Data Analysis and Visualization

## Part 4: Methods and Plan